### PR TITLE
Add datastore helper ftns

### PR DESCRIPTION
**Pull request by samup4web**
## Description
* Adds new helper function to read data of interest from database.
* Creates new sql database table for storing AccessPoint data
* Minor refactoring
* Updates UT
 

## Solved issue(s)
* #61 
* #62

### CODE REVIEW

File: server/include/data_store.hpp [hunk 1]
```diff
@@ -43,24 +41,30 @@ class DataStore
 
     bool AssignDeviceToEmployee(const std::string& device_id, const std::string& employee_id);
 
-    bool InsertRSSIReadings(const std::string& device_id,
-                            std::vector<std::pair<std::string, double>> macaddr_rssi_datapoints);
+    bool InsertRSSIReadings(const std::string& device_id, std::vector<AccessPointRssiPair> accesspoint_rssi_pair_list);
```
Comment (platisd) [inline server/include/data_store.hpp:44]: > AccessPointRssiPair

Cool! I was also thinking of something like that for the ins-node.

File: server/src/data_store.cpp [hunk 2]
```diff
@@ -206,6 +222,7 @@ bool DataStore::RunQuery(const std::string& sql)
 
 int DataStore::DbCallback(void* not_used, int argc, char** argv, char** azColName)
 {
+    (void)not_used;
```
Comment (platisd) [inline server/src/data_store.cpp:225]: Is this variable for future-proofing the interface?

File: server/test/mocks/mock_data_store.hpp [hunk 3]
```diff
@@ -24,15 +24,21 @@ class MockDataStore : public DataStore
...
     MOCK_METHOD1(ClearDeviceTable, bool(const std::string&));
 
-    MOCK_METHOD1(ReadDistinctMacAddrs, std::vector<std::string>(const std::string&));
+    MOCK_METHOD1(GetDistinctAccessPoints, std::vector<AccessPoint>(const std::string&));
+
+    MOCK_METHOD2(GetRSSISeriesData, std::vector<int32_t>(const std::string&, AccessPoint));
+
+    MOCK_METHOD2(GetRSSISeriesData, std::vector<AccessPointRssiListPair>(const std::string&, std::vector<AccessPoint>));
+
+    // MOCK_METHOD2()
```
Comment (platisd) [inline server/test/mocks/mock_data_store.hpp]: Forgot to delete or something?

**Review by platisd — DISMISSED** [review]

File: server/src/data_store.cpp (hunk 2 above)
Comment (samup4web) [reply server/src/data_store.cpp:225]: Sqlite3 requires this callback function with this signature for some of its operation. It only happens the variable is not used in this case. Actually, the name is a bad choice copied from examples.

File: server/test/mocks/mock_data_store.hpp (hunk 3 above)
Comment (samup4web) [reply server/test/mocks/mock_data_store.hpp]: Yes, I will remove.

**Review by platisd — APPROVED** [review]

**Pull request merged**